cpp/oneapi/dal/algo/knn/backend/gpu/train_kernel_brute_force_dpc.cpp

```diff
@@ -67,7 +67,7 @@ static train_result call_daal_kernel(const context_gpu& ctx,
 
     daal::algorithms::classifier::ModelPtr model_ptr(new daal_knn::Model(column_count));
     if (!model_ptr) {
```

Comment:
Why you think that `model_ptr` will be `nullptr` when memory allocation is failed? I think `new daal_knn::Model(column_count)` would throw `std::bad_alloc` if `operator new` is not overloaded. Is it overloaded by DAAL in the way it simply returns `nullptr` in case of bad allocation?

Reply:
fixed

Reply:
@mchernov-intel so what's the conclusion? I see that you changed this behavior, but why? I just asked you if you are sure you are doing right. Have you checked that new operator used in `Model` really throws `bad_alloc`? I remember that for DAAL objects `new` and `delete` are overloaded.

Reply:
resolved via status